docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,331 @@
+    type: entity
+    extends: entity.my_entity
+    attributes:
+      - ref: new.attribute.name
+        requirement_level: opt_in
+        role: descriptive
+```
+
+Notes:
+
+- You cannot change the `name` or `type` fields of the new entity.
+- You cannot change the set of identifying attributes.
+
+## FAQ
+
+### When to define a new entity?
+
+There are two scenarios where entities should be defined:
+
+- When you are generating a new signal (log, metric, span, etc.)
+  and no existing entity makes sense as the "source".
+- (future) When you need to describe an entity hierarchy from
+  some system of record (e.g. resources in kubernetes, assets
+  in a cloud).
+
+For example, if a new clustering solution (e.g. Hashicorp's Nomad) is
+defined, and existing container-based entities are not enough, then
+new entities should be defined.
+
+### What is an "is-a" relationship?
+
+OpenTelemetry, as an open ecosystem, cannot understand and model all
+possible entities that exist in the world. Instead, we are allowing
+overlapping definitions across domains. For example, the `container` and
+`k8s.container` entities exist, and generally every `k8s.container` is
+a `container`, but not every `container` runs in kubernetes.
+
+An "is-a" relationship denotes that one entity is describing the exact
+same system component as another entity, but from a different domain. In
+the above example, `k8s.container` models containers from the kubernetes
+domain, while `container` is a general model for containers, regardless
+of how they are run (e.g. podman, docker, kubernetes, FAAS, etc.)
+
+"is-a" relationships denote this relationship in entities allowing
+OpenTelemetry to fully model a subset of entities (e.g. all known
+`k8s` resources as entities), but still allow the extended ecosystem
+to grow and evolve with new entities in the future.
+
+### When to define an "is-a" relationship vs. extending descriptive attributes?
+
+There are two key rules:
+
+- Default to introducing separate entities with a clear "is a" (or similar) relationship
+- Extend an entity with new descriptive attributes if, and only if, the following is true:
+  - The extending entity cannot be associated with any telemetry by itself.
+    - Example 1: When adding a `windows.process` entity, you do not expect
+      to create any specific process metrics or logs that would be
+      specific to `windows.process`, instead all data is still reported
+      against `process` entity.
+    - Example 2: When adding a `docker.container` entity, you do not
+      expect to create any specific container signals, instead logs,
+      metrics and spans would be reported against the `container` entity.
+  - The extending entity doesn’t have any other entities that logically can
+    only be associated with it.
+    *Note: this only applies to entities as signals with relationships.*
+
+This avoids the complexities of subtyping and ambiguous attribute usage.
+
+### How to define identifying attributes?
```

Comment:
Should we also say that the set of identifying attributes defines the lifespan of an entity. The values cannot change during the lifespan.